Devscord.DiscordFramework/Middlewares/Factories/ChannelContextFactory.cs

```diff
@@ -1,14 +1,19 @@
 ﻿using Devscord.DiscordFramework.Middlewares.Contexts;
-using Discord.Rest;
-using Discord.WebSocket;
+using Discord;
+using System;
 
 namespace Devscord.DiscordFramework.Middlewares.Factories
 {
-    internal class ChannelContextFactory : IContextFactory<IRestMessageChannel, ChannelContext>
+    internal class ChannelContextFactory : IContextFactory<IMessageChannel, ChannelContext>
     {
-        public ChannelContext Create(IRestMessageChannel restChannel)
+        public ChannelContext Create(IMessageChannel messageChannel)
         {
-            return restChannel == null ? null : new ChannelContext(restChannel.Id, restChannel.Name);
+            if (messageChannel == null)
+            {
+                throw new ArgumentNullException();
```

Comment:
tak w sumie to próba użycia `messageChannel.Id` też wywali wyjątek 

no chyba że ma wywalić akurat ten wyjątek